internal/dataplane/client/utils.go

```diff
@@ -31,60 +31,57 @@ import (
 // CompileUDPRouteToDataPlaneBackend takes a UDPRoute and the Gateway it is
 // attached to and produces Backend Targets for the DataPlane to configure.
 func CompileUDPRouteToDataPlaneBackend(ctx context.Context, c client.Client, udproute *gatewayv1alpha2.UDPRoute, gateway *gatewayv1beta1.Gateway) (*Targets, error) {
-	// TODO: add support for multiple rules https://github.com/Kong/blixt/issues/10
-	if len(udproute.Spec.Rules) != 1 {
-		return nil, fmt.Errorf("currently can only support 1 UDPRoute rule, received %d", len(udproute.Spec.Rules))
-	}
-	rule := udproute.Spec.Rules[0]
-
-	// TODO: add support for multiple rules https://github.com/Kong/blixt/issues/10
-	if len(rule.BackendRefs) != 1 {
-		return nil, fmt.Errorf("expect 1 backendRef received %d", len(rule.BackendRefs))
-	}
-	backendRef := rule.BackendRefs[0]
 
-	gatewayIP, err := getGatewayIP(gateway)
+	gatewayIP, err := GetGatewayIP(gateway)
 	if gatewayIP == nil {
 		return nil, err
 	}
 
-	gatewayPort, err := getGatewayPort(gateway, udproute.Spec.ParentRefs)
+	gatewayPort, err := GetGatewayPort(gateway, udproute.Spec.ParentRefs)
 	if err != nil {
 		return nil, err
 	}
-
-	// TODO only using one endpoint for now until https://github.com/Kong/blixt/issues/10
-	var target *Target
-	if udproute.DeletionTimestamp == nil {
-		endpoints, err := endpointsFromBackendRef(ctx, c, udproute.Namespace, backendRef)
-		if err != nil {
-			return nil, err
-		}
-
-		for _, subset := range endpoints.Subsets {
-			if len(subset.Addresses) < 1 {
-				return nil, fmt.Errorf("addresses not ready for endpoints")
-			}
-			if len(subset.Ports) < 1 {
-				return nil, fmt.Errorf("ports not ready for endpoints")
-			}
-
-			if subset.Addresses[0].IP == "" {
-				return nil, fmt.Errorf("empty IP for endpoint subset")
+	var backendTargets []*Target
+	for _, rule := range udproute.Spec.Rules {
+		for _, backendRef := range rule.BackendRefs {
+			endpoints, err := endpointsFromBackendRef(ctx, c, udproute.Namespace, backendRef)
+			if err != nil {
+				return nil, err
 			}
 
-			ip := net.ParseIP(subset.Addresses[0].IP)
-
-			podip := binary.BigEndian.Uint32(ip.To4())
-
-			target = &Target{
-				Daddr: podip,
-				Dport: uint32(subset.Ports[0].Port),
+			for _, subset := range endpoints.Subsets {
+				if len(subset.Addresses) < 1 {
+					return nil, fmt.Errorf("addresses not ready for endpoints")
+				}
+				if len(subset.Ports) < 1 {
+					return nil, fmt.Errorf("ports not ready for endpoints")
+				}
+
+				for _, addr := range subset.Addresses {
+					if addr.IP == "" {
+						return nil, fmt.Errorf("empty IP for endpoint subset")
+					}
+
+					ip := net.ParseIP(addr.IP)
+
+					podip := binary.BigEndian.Uint32(ip.To4())
```

Comment:
There's a couple places where we implicitly assume V4, this is totally fine for now, but if we don't have an issue tracking adding V6 support we should create that.

Reply:
https://github.com/kubernetes-sigs/blixt/issues/120